pennylane/devices/experimental/device_api.py

```diff
@@ -168,6 +178,19 @@ def shots(self) -> Shots:
         """
         return self._shots
 
+    @property
+    def wires(self) -> Wires:
+        """The device wires.
+
+        Note that wires are optional, and the default value of None means any wires can be used.
+        If a device has wires defined, they will only be used for certain features. This includes:
+
+        * Validation of tapes being executed on the device
```

Comment:
If wires are to be used for validation, that might be worth adding in this PR. `preprocess` can include that step.

Reply:
I was hoping to do it in a separate PR today because I fear that it'll be a bit more involved, but what the heck why not

Reply:
I think its good to have this PR focusing on the "Interface" aspect of wires. Then we can have a follow-up PR focusing on the "implementation" part of wires.